### PR TITLE
Avoid duplicating rail/road/tramtype table code

### PR DESCRIPTION
**Pull request by FLHerne**
All this virtually-identical code was copied in triplicate.

Should be no functional changes; regression tests pass.

### CODE REVIEW

**Comment by FLHerne** [comment]
Variable names `table_prop_id` and `cond_tracktype_not_defined` could do with review, I wasn't really sure what to call them.

(I think the current names were suggested by someone else already)

**Comment by andythenorth** [comment]
Tests pass for me. 
NML example_roadtype_and_tramtype.nml builds.
Iron Horse builds.

**Comment by glx22** [comment]
Since checks are enabled, a rebase to master is required to include workflow file.

File: nml/ast/tracktypetable.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    table_prop_id = 0x16
+    cond_tracktype_not_defined = 0x0F
+
+    def __init__(self, *args, **kwargs):
+        global_constants.is_default_roadtype_table = False
+        super().__init__(*args, **kwargs)
+
+
+class TramtypeTable(BaseTracktypeTable):
+    track_kind = 'road'
```
Comment (glx22) [inline nml/ast/tracktypetable.py]: maybe
```suggestion
    track_kind = 'tram'
```

**Pull request merged**